Newsletters: add 219 (2022-09-28) (#867)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2022-09-28-newsletter.md

---
title: 'Bitcoin Optech Newsletter #219'
permalink: /en/newsletters/2022/09/28/
name: 2022-09-28-newsletter
slug: 2022-09-28-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME:harding

## News

- **LN fee ratecards:** Lisa Neigut [posted][neigut ratecards] to the
  Lightning-Dev mailing list a proposal for *fee ratecards* that would
  allow a node to advertise four tiered rates for forwarding fees.  For
  example, a forwarded payment that leaves a channel with less than 25%
  of its outbound capacity available to forward subsequent payments
  would need to pay proportionally more than a payment which leaves 75%
  of its outbound capacity available.

    Developer ZmnSCPxj [described][zmnscpxj ratecards] a simple way to
    use ratecards, "you can model a rate card as four separate channels
    between the same two nodes, with different costs each.  If the path

>     at the lowest cost fails, you just try at another route that may

Reviewer: ```suggestion
    at the lowest cost fails, you just try another route that may
```

    have more hops but lower effective cost, or else try the same
    channel at a higher cost."

    The proposal also allows for negative fees.  For example, a channel
    could subsidize payments forwarded through it when it had more than

>     75% outbound capacity by adding 1 satoshi to every 1,000 satoshi in

Reviewer: ```suggestion
    75% outbound capacity by adding 1 satoshi to every 1,000 satoshis in
```

    payment value.  Negative fees could be used by merchants to
    incentivize others to restore inbound capacity to channels
    frequently used for receiving payments.

    Neigut notes that some nodes are already adjusting their fees based

>     on channel capacity, and that fee ratecards provides a more

Reviewer: ```suggestion
    on channel capacity, and that fee ratecards provide a more
```
Or maybe "and that fee ratecards feature provides a more ..."

    efficient way to communicate their intentions to the network than
    frequently advertising new feerates via the LN gossip network.

- **Bitcoin implementation designed for testing soft forks on signet:**
  Anthony Towns [posted][towns bi] to the Bitcoin-Dev mailing list a
  description of a fork of Bitcoin Core he's working on that will only
  operate on the default [signet][topic signet] and which will enforce

>   rules for soft fork proposals with high quality specification and

Reviewer: ```suggestion
  rules for soft fork proposals with high-quality specifications and
```

  another similar change (e.g. comparing [OP_CHECKTEMPLATEVERIFY][topic
  op_checktemplateverify] to [SIGHASH_ANYPREVOUT][topic
  sighash_anyprevout]).  Towns also plans to include proposed major
  changes to transaction relay policy (such as [package relay][topic
  package relay]).

    Towns is seeking constructive criticism of the idea for this new
    testing implementation, called [Bitcoin Inquisition][], as well as
    reviews of the [pull requests][bi prs] adding the initial set of
    soft forks to it.

## Selected Q&A from Bitcoin Stack Exchange

*[Bitcoin Stack Exchange][bitcoin.se] is one of the first places Optech
contributors look for answers to their questions---or when we have a
few spare moments to help curious or confused users.  In
this monthly feature, we highlight some of the top-voted questions and
answers posted since our last update.*

{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

- [Is it possible to fingerprint HD wallets?]({{bse}}115311)
  Pieter Wuille points out that while identifying UTXOs created using an [HD

>   wallet][topic bip32] is not possible, other onchain data can be used to

Reviewer: ```suggestion
  wallet][topic bip32] is not possible, other on-chain data can be used to
```

Author: We previously settled on "onchain", see https://github.com/bitcoinops/bitcoinops.github.io/blob/master/STYLE.md#compound-words

>   fingerprint wallet software including type of inputs used, types of outputs

Reviewer: ```suggestion
  fingerprint wallet software including types of inputs used, types of outputs
```

  created, order of inputs and outputs in the transaction, [coin

>   selection][topic coin selection] algorithm, and use of [timelocks][topic timelocks].

Reviewer 2: Reworded the question, but took some liberties

> - [Why is there a 5 day gap between the genesis block and block 1?]({{bse}}115344)

Reviewer: ```suggestion
- [Why is there a 5-day gap between the genesis block and block 1?]({{bse}}115344)
```

  Murch notes that the gap in the timeline could be explained by the genesis
  block having a higher difficulty target than required, Satoshi setting the
  block's timestamp in the past, or the [original Bitcoin software waiting][github
  jrubin annotated] for a peer before continuing to mine.

- [Is it possible to set RBF as always-on in bitcoind?]({{bse}}115360)
  Michael Folkson and Murch explain the `walletrbf` configuration option and
  list a series of related changes involving defaulting to [RBF][topic rbf] in the
  GUI, defaulting to RBF in RPCs, and using [`mempoolfullrbf`][news208
  mempoolfullrbf] to allow replacements without signaling.

> - [Why would I need to ban peer nodes in Bitcoin network?]({{bse}}115183)

Reviewer: ```suggestion
- [Why would I need to ban peer nodes in the Bitcoin network?]({{bse}}115183)
```

  In contrast to [discouraging a peer][bitcoin 23.x banman], user RedGrittyBrick
  explains that a node operator could choose to manually ban a peer using the
  [`setban`][setban rpc] RPC if the peer is misbehaving, a suspected malicious or surveillance

>   node, or part of cloud provider's network, among other reasons.

Reviewer: ```suggestion
  node, or part of a cloud provider's network, among other reasons.
```

Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*

- [Bitcoin Core #26116][] allows the `importmulti` RPC to import a
  watch-only item even if the wallet's private keys are encrypted.  This
  matches the behavior of the old `importaddress` RPC.

- [Core Lightning #5594][] makes several changes, including adding and
  updating several APIs, to allow the `autoclean` plugin to delete old
  invoices, payments, and forwarded payments.

- [Core Lightning #5315][] allows the user to choose the *channel
  reserve* for a particular channel.  The reserve is the amount a node
  will normally not accept from a channel peer as part of a payment or
  forward.  If the peer later attempts to cheat, the honest node can
  spend the reserve.  Core Lightning defaults to a channel reserve of 1%
  of the channel balance, penalizing by that amount any peers that
  attempt to cheat.

    This merged PR allows the user to reduce the reserve for a specific
    channel, including reducing it to zero.  Although this can be
    dangerous---the lower the reserve, the less penalty there is for
    cheating---it can be useful for certain situations.  Most notably,

>     setting the reserve can allow the remote peer to spend all of their

Reviewer: (check me on this!)
```suggestion
    setting the reserve to zero can allow the remote peer to spend all of their
```

{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

- [Is it possible to fingerprint HD wallets?]({{bse}}115311)
  Pieter Wuille points out that while identifying UTXOs created using an [HD
  wallet][topic bip32] is not possible, other onchain data can be used to
  fingerprint wallet software including type of inputs used, types of outputs
  created, order of inputs and outputs in the transaction, [coin
  selection][topic coin selection] algorithm, and use of [timelocks][topic timelocks].

- [Why is there a 5 day gap between the genesis block and block 1?]({{bse}}115344)
  Murch notes that the gap in the timeline could be explained by the genesis
  block having a higher difficulty target than required, Satoshi setting the
  block's timestamp in the past, or the [original Bitcoin software waiting][github

>   jrubin annotated] for a peer before continuing to mine.

Reviewer 3: ```suggestion
  jrubin annotated] for a peer before starting to mine.
```
